CMakeLists.txt

```diff
@@ -22,7 +22,12 @@ endif()
 if(POLICY CMP0065)
   cmake_policy(SET CMP0065 NEW)
 endif()
-
+if(POLICY CMP0051)
+  # cmake 3.1 and higher include generator expressions in SOURCES property,
+  # but i don't want to bother with stripping off the TARGET_OBJECTS elements
+  # from it. so let's stick with the old behavior now.
+  cmake_policy(SET CMP0051 OLD)
+endif()
```

Comment:
okay, it took me a while to realize this was related to `get_target_property(sources ${target} SOURCES)` in `maybe_add_boost_dep()`. could you mention `maybe_add_boost_dep()` in this comment?